src/sentry/incidents/logic.py

```diff
@@ -25,6 +34,7 @@ def create_incident(
     projects=None,
     groups=None,
 ):
+    assert status in (IncidentStatus.CREATED, IncidentStatus.DETECTED)
```

Comment:
If a user were to create a closed incident wouldn't this result in a 500? This seems like it should be a 40x error.

Reply:
I mostly put this in place to catch people accidentally using this function incorrectly while writing new code, since at the moment the status is hardcoded to `IncidentStatus.CREATED` in the creation api. 

We haven't actually specified how automatic detection will feed into Sentry at all, I'm not sure whether they'll be using an API, or we'll be consuming from a queue somewhere, etc. So we might not need to do any user validation of the creation status.